setuptools/dist.py

```diff
@@ -83,6 +84,24 @@ def _read_list(name):
             return None
         return values
 
+    def _read_long_description():
```

Comment:
I'd like to avoid copy/pasting this metadata reading code. Let's instead get `importlib_metadata` vendored in here and rely on that to read metadata.

Reply:
I'm not sure this would solve it. `imporlib_metadata` doesn't really parse the metadata. It just returns an email message object which can then be inspected. We would still need to do most if not all of these steps.